pythonforandroid/recipes/python3/patches/fix-ctypes-util-find-library.patch

```diff
@@ -10,9 +10,15 @@ diff --git a/Lib/ctypes/util.py b/Lib/ctypes/util.py
 +if True:
 +    def find_library(name):
 +        # Check the user app libs and system libraries directory:
-+        app_root = os.path.normpath(os.path.abspath('../../'))
-+        lib_search_dirs = [os.path.join(app_root, 'lib'), "/system/lib"]
-+        for lib_dir in lib_search_dirs:
++        from jnius import autoclass
++        activity = autoclass('org.kivy.android.PythonActivity').mActivity
++        package_name = activity.getApplicationContext().getPackageName()
++        manager = activity.getApplicationContext().getPackageManager()
++        nativeLibDir = manager.getApplicationInfo(
++            package_name, 1024  # PackageManager.GET_SHARED_LIBRARY_FILES
```

Comment:
why don't you `manager.GET_SHARED_LIBRARY_FILES` rather than hardcoding that `1024` digit?
I guess you should be able to use that `manager` instance right? Otherwise maybe something like:
```python
PackageManager = autoclass('android.content.pm.PackageManager')
PackageManager.GET_SHARED_LIBRARY_FILES
```

Reply:
good point I'll test out a fix. Interestingly `manager.GET_SHARED_LIBRARY_FILES` is 0 which is probably a pyjnius bug, but using the class works